### ISSUE TITLE
Character/Clones -> "active" flag in /characters/{character_id}/clones/

### ISSUE THREAD

**Issue by Legedric** [ccpgames/esi-issues#94]
Although there is an endpoint to get a character's clones from ESI there seems to be no flag about which clone is currently active and we are unable to get the currently active implants for a character to calculate skill training times.

Adding an "active" flag to "/characters/{character_id}/clones/" like this should be sufficient:
`{
  "home_location": {
    "location_id": 1021348135816,
    "location_type": "structure"
  },
  "jump_clones": [
    {
      "implants": [
        22118
      ],
      "location_id": 60003463,
      "location_type": "station",
      "active": true
    },
    {
      "implants": [],
      "location_id": 1021348135816,
      "location_type": "structure",
      "active": false
    }
  ]
}`

**Comment by Regner** [comment]
I thought your active clone wasn't listed in the all clones.

**Comment by ddouglas493** [comment]
Its not. The only difference between clones is the implants that are currently plugged in. Just take into account the current implants that are plugged in and you'll do fine.

**Comment by Legedric** [comment]
@Regner Yes I know that.
But instead of adding implants to the character endpoint (where they wouldn't belong technically if you ask me) it would be straight forward to add the active clone to the clones endlpoint including a simple "active" flag.

**Comment by Regner** [comment]
OK, so what you're really asking for is for the two resources to be merged (current character implants and clones) with a flag to specify which is active.

**Comment by Legedric** [comment]
@Regner Perhaps I am missing something, but where do I see the current character implants on ESI? Because talking of a "merge" I would assume they already exist somwhere there, right? Or are you referring to the old structure from EVE's XML API CharacterSheet where the current implants were just a list within the base sheet?

For me technically I expect the clones endpoint to provide all clones of that character back to me in its response and this of course includes the active clone. Although I don't know if that fits the structure of EVE's database :)

Of course it wouldn't be right to just put it into the "jump_clones" list so instead of an active flag it would also be possible to just create another JSON object "current_clone" in the response with the same structure as every jump clone entry has (including its implants, location and location type).

**Comment by 7sempra** [comment]
Ping on this. Is there currently *any* way to read a character's active implants? I personally don't care where they appear, as long as they're available somewhere.

**Comment by Robbilie** [comment]
the mobile app shows implants right?

**Comment by Robbilie** [comment]
https://esi.tech.ccp.is/latest/#!/Clones/get_characters_character_id_clones

**Comment by 7sempra** [comment]
That ESI endpoint only shows jump clones, not the implants in the character's *current* clone.

It seems like this should really be part of some character sheet endpoint and not in jump clones. What we really want to know is the current clone's attributes (i.e. base attrs + implants). Otherwise we can't do training time estimations.

**Comment by Robbilie** [comment]
yeah i feared that, but i still think it should be in the clones endpoint bc its obviously a clone too that you are currently in.

so it would be good if CCP could add the active clone to that list

**Comment by MandorSawall** [comment]
Is it possible that we still don't have a way of getting the current implants?

**Comment by Legedric** [comment]
Still no way to get current active set of implants...
Still no way of getting clone states...

In short: ESI is still (after Ascension got released in Nov 2016) useless in terms of skill calculations...

**Comment by MandorSawall** [comment]
Is there this support in CREST (I just started using eve apis :-)

**Comment by Legedric** [comment]
I am not working with CREST, but according to the docs I would say no:
http://eveonline-third-party-documentation.readthedocs.io/en/latest/crest/character/char_character.html

The XML API is at least able to list the current active set of implants but not the clone state (alpha / omega)

**Comment by whinis** [comment]
The XML api can return clone state (alpha/omega) through account based endpoints that CCP has stated are not coming to ESI.

**Comment by MandorSawall** [comment]
What? Isn't ESI one API to replace them all? 
We can manage fleet, send emails, update structure vulnerability schedule but current implants and clone state? No no, that is too much info!

**Comment by Legedric** [comment]
Well, I am afraid but that's how it is. And the fact that this issue (which is highly important for any skill trainer out there) is open and pretty much untouched since Nov 2016 (4 months +) is not making a good impression about what to expect from ESI in the near future in regards of a full scope replacement for XML/CREST.

**Comment by MandorSawall** [comment]
Is there some info about this? CCP does not want us to have this information or are they just lazy?
I'd like the FC to see player current implants so he can suggest getting some if fleet members don't have them. I was hoping for current fits too, but I can see how that could be a problem. Also, ship names are not available?

**Comment by Legedric** [comment]
So before we start micing things up here:
**1. Current active clone including its implants:**
CCP may add this sometimes... Who knows, they did not say they won't so far. But who knows when and if at all.

**2. Clone state (Alpha/Omega)** 
As you can see in this issue (and porbably several others), CCP won't add account information to the API anymore:
https://github.com/ccpgames/esi-issues/issues/95
They are saying at least they plan to add something to see the clone state but make it not that obvious (whatever that means). But as this issue is also 5 months old and nothing happened into that direction I doubt that CCP is still on track with this at all.

**Comment by zweizeichen** [comment]
I can understand that you are eagerly awaiting these features, you voiced your opinion +1'ing und complaining won't make things go any faster. There are 170 open issues right now, be patient. If CCP would not want to add this feature this issue would be closed by now. Remember where we came from with CREST and the XML API a couple of years ago. The amount of support and communication we're getting now is tremendous compared to back then. It is not like someone will switch off your XML API tomorrow or something. Just give it some time.

**Comment by MandorSawall** [comment]
I haven't used previous APIs. It just looked like this is dead. Months have passed since. I need (ok, would be helpful) to see charcter implants for public fleets as well as ship fitting. If it's some security thing I'd just put permissions in red colour and make sure users know how to revoke permissions themselves.

**Comment by snitchashor** [comment]
It's quite easy: If those information is available through CREST/XML it has priority expect it soom(tm), if not, it doesn't.

**Comment by Kyria** [comment]
you can still use the XML API using SSO with the right scopes to get informations you need (and in this case, the current clone implants). 
FYI: http://eveonline-third-party-documentation.readthedocs.io/en/latest/xmlapi/authentication.html#character-access

**Comment by aquarhead** [comment]
To make it clear, ESI's [`/clones`](https://esi.tech.ccp.is/latest/#!/Clones/get_characters_character_id_clones) endpoint only returns a character's jump clones, which does not include your active clone. (The documentation/spec should be updated to clarify this)

Closing this and the request for adding current clone's implants information will be merged into #15
